### PR TITLE
fix(vscode): show upstream inference cost for OpenRouter BYOK sessions

### PR DESCRIPTION
**Pull request by iscekic**
> **kwf trial run.** This PR came from a workflow trial request, not from a tracked work item. Review it as you would any other PR; the label `kwf-trial` marks where it came from.

## Request

> Surface: the VS Code extension (Kilo-Org/kilocode issue 13829).
> 
> OpenRouter BYOK sessions always show $0, because providerCost reads `usage.cost` and ignores `usage.cost_details.upstream_inference_cost`.
> 
> OpenRouter returns, for every response it routes through a BYOK provider key:
> 
>     "usage": { "cost": 0, "is_byok": true,
>                "cost_details": { "upstream_inference_cost": 0.00000445 } }
> 
> `cost` is what OpenRouter charged the account, which is 0 by definition for BYOK routing. The real spend is `cost_details.upstream_inference_cost`. Reported on 7.5.14 and 7.5.15 with `openrouter/z-ai/glm-5.3-flash`.
> 
> Expected: a BYOK session shows the upstream inference cost, and a non-BYOK session keeps showing `usage.cost` exactly as it does today.
> 
> Rule: reproduce first. If it does not reproduce on main, answer no_change with the evidence.

## Changelog for users

- OpenRouter BYOK sessions in the VS Code extension no longer show $0: the session cost now reports `usage.cost_details.upstream_inference_cost` — the real spend on the user's own provider key — whenever OpenRouter bills the account $0 or only its routing fee for BYOK routing (Kilo-Org/kilocode#13829).
- Non-BYOK OpenRouter sessions are unchanged: the displayed cost is still `usage.cost`, exactly as today.

## Changelog for maintainers

- The OpenRouter chat-completions branch of the session `providerCost` helper now prefers `costDetails.upstreamInferenceCost` over the billed `cost` whenever the upstream figure is strictly greater; previously upstream was preferred only for the Kilo provider. That one expression is the whole fix — start the review there.
- Non-BYOK safety: a non-BYOK response always bills the account at least the upstream cost, so `upstream > regular` is false and `usage.cost` is returned untouched; unit tests pin both the BYOK and non-BYOK payload shapes.
- Edge case to review: when a response carries `costDetails` but no top-level `cost`, the new expression (`regular ?? -Infinity`) now returns the upstream figure, where the old code fell through to the token-based estimate. Only reachable if OpenRouter ever omits `cost` while still reporting cost details; not covered by a test.
- New unit suite for the OpenRouter chat-completions path covers five cases: BYOK with `cost: 0`, BYOK where `cost` is only the routing fee, non-BYOK unchanged, no `costDetails` unchanged, and the Kilo-provider preference.
- The transport doc comment in the session cost path now documents the BYOK semantics (account billed $0 or a routing fee; true spend on the user's own key).
- Verification: the reproduce-first rule was proved live; the run's decisive log excerpts are appended below under their own heading.

## E2E proof — log excerpts

```
[e1] Rule: reproduce first. If it does not reproduce on main, answer no_change w -> pass :: Baseline (e1-baseline.log) fails with Expected: 0.00000445 Received: 0 and Expected: 0.06418625 Received: 0.0032093125; packed (e1-packed.log) reports 10 pass a
```

<details><summary><code>/Users/igor/.local/share/kwf/sections/surface-the-vs-code-extensio-5184/e2e-cli/e1-baseline.log</code></summary>

```
155 |     expect(result.cost).toBe(0.00000445)
                              ^
error: expect(received).toBe(expected)
Expected: 0.00000445
Received: 0
      at <anonymous> (/Users/igor/.local/share/kwf/wt/surface-the-vs-code-extensio-5184/packages/opencode/test/kilocode/provider-cost.test.ts:155:25)
(fail) KiloSession.providerCost — OpenRouter chat completions > uses upstream_inference_cost when OpenRouter reports $0 for BYOK routing [1.51ms]
169 |           },
170 |         },
171 |       },
172 |     })
173 |
174 |     expect(result.cost).toBe(0.06418625)
                              ^
error: expect(received).toBe(expected)
Expected: 0.06418625
Received: 0.0032093125
      at <anonymous> (/Users/igor/.local/share/kwf/wt/surface-the-vs-code-extensio-5184/packages/opencode/test/kilocode/provider-cost.test.ts:174:25)
(fail) KiloSession.providerCost — OpenRouter chat completions > uses upstream_inference_cost when OpenRouter bills only its BYOK routing fee [3.19ms]
 3 pass
 5 filtered out
 2 fail
 5 expect() calls
Ran 5 tests across 1 file. [2.15s]
```
</details>

<details><summary><code>/Users/igor/.local/share/kwf/sections/surface-the-vs-code-extensio-5184/e2e-cli/e1-packed.log</code></summary>

```
bun test v1.4.0 (34cbb9a40)
 10 pass
 0 fail
 10 expect() calls
Ran 10 tests across 1 file. [2.10s]
```
</details>

### CODE REVIEW

File: packages/opencode/src/kilocode/session/index.ts [hunk 1]
```diff
@@ -221,8 +228,13 @@ export namespace KiloSession {
...
       // prefer the upstream cost (the user's true spend). For the OpenRouter provider
-      // itself, the regular `cost` field is what the user is billed.
-      const cost = isKilo && upstream !== undefined ? upstream : regular
+      // itself, the regular `cost` field is what the user is billed — except when the
+      // request routes through a BYOK provider key: then OpenRouter bills the account
+      // $0 (or only its routing fee) and the true spend is the upstream inference cost
+      // charged to the user's own key. A non-BYOK response always bills at least the
+      // upstream cost, so preferring upstream only when it exceeds the billed amount
+      // never changes non-BYOK sessions.
+      const cost = upstream !== undefined && (isKilo || upstream > (regular ?? -Infinity)) ? upstream : regular
```
Comment (chrarnoldus) [inline packages/opencode/src/kilocode/session/index.ts]: I think technically it would have to be `upstream+regular` for OpenRouter BYOK

**Review by chrarnoldus — APPROVED** [review]
